python/tchannel/tornado/data.py

```diff
@@ -0,0 +1,294 @@
+# Copyright (c) 2015 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import tornado
+import tornado.gen
+
+from .stream import InMemStream
+from .util import get_arg
+from .util import deserialize_default
+from .util import serialize_default
+
+from ..exceptions import TChannelException
+from ..messages.common import FlagsType
+from ..messages.common import StreamState
+from ..zipkin.trace import Trace
+
+
+class Request(object):
+    """Represents an incoming request to an endpoint.
+
+    Request class is used to represent the CallRequestMessage at User's level.
+    This is going to hide the protocol level message information.
+    """
+
+    # TODO decide which elements inside "message" object to expose to user.
+    def __init__(
+            self,
+            id=None,
+            flags=FlagsType.none,
+            ttl=10,
+            tracing=None,
+            service=None,
+            headers=None,
+            checksum=None,
+            argstreams=None,
+            serializer=None,
+            deserializer=None,
+    ):
+        self.flags = flags
+        self.ttl = ttl
+        self.service = service
+        self.tracing = tracing or Trace()
+        # argstreams is a list of InMemStream/PipeStream objects
+        self.argstreams = argstreams
+        self.checksum = checksum
+        self.id = id
+        self.headers = headers or {}
+        self.state = StreamState.init
+        self.endpoint = ""
+        self.header = None
+        self.body = None
+
+        self.serializer = serializer or serialize_default
```

Comment:
The default serializer will just return what we pass.
In that case, I don't need if-else in the code.